cmake/modules/SetRPATH.cmake

```diff
@@ -0,0 +1,15 @@
+set(CMAKE_SKIP_BUILD_RPATH FALSE)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+
+# Append non-standard external dependency directories
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+# CMake 3.25 has a better scoping solution: 'block()'
+function(SetRPATH)
+  set(p "${CMAKE_INSTALL_FULL_LIBDIR}")
+  list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${p}" i)
+  if("${i}" STREQUAL "-1")
+    set(CMAKE_INSTALL_RPATH "${p}" PARENT_SCOPE)
+  endif()
+endfunction()
+SetRPATH()
```

Comment:
I'm not sure if we need a separate function.

Reply:
I would have much preferred to write:

<pre>
block()
  set(p "${CMAKE_INSTALL_FULL_LIBDIR}")
  list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${p}" i)
  if("${i}" STREQUAL "-1")
    set(CMAKE_INSTALL_RPATH "${p}" PARENT_SCOPE)
  endif()
endblock()
</pre>

However, the CMake language only got [anonymous] block scoping in version `3.25`, while this project currently sets `3.7` as the minimum version; I cannot use the `block()` construct.

This is why I added the comment:

    # CMake 3.25 has a better scoping solution: 'block()'

My hope was that it would be a hint to some future janitor to use the `block()` construct when available.

Reply:
There are only five relevant Iines in that block and I think it's clearer to add them directly in CMakeLists.txt without adding an additional file.

Reply:
Actually, it would be much clearer to move *even more stuff* into their own files, but I guess we have different tastes. I will move it all into `CMakeLists.txt`, as that is what you prefer.